server/aesdsocket.c

```diff
@@ -0,0 +1,325 @@
+        // perror("Daemon Started\n");
+    }
+
+    /*Server Socket Listening to Connections*/
+    int listenRetVal = listen(serverSocket, 5);
+
+    if (listenRetVal < 0)
+    {
+        perror("[-] Listen Failed\n");
+    }
+
+    // /*Debug Steps*/
+    // printf("Server Address : %d\n", serverAddress.sin_addr.s_addr);
+    // printf("Server Family : %d\n", serverAddress.sin_family);
+    // printf("Server Port : %d\n", htons(serverAddress.sin_port));
+
+    // char *receiveFromClient;
+    char *writebuffer;
+
+    int totalBuffer = BUFFER_STD_SIZE;
+    int currentSize = 0;
+
+    char receiveFromClient[BUFFER_STD_SIZE];
+
+    int bytesToBeWritten = 0;
+    int readBufferSize = 0;
+
+    char *readFromFile;
+
+    int bytesToBeRead = 0;
+
+    int lseekValue = 0;
+
+    fd = open("/var/tmp/aesdsocketdata", O_CREAT | O_RDWR | O_TRUNC, 0644);
+
+    while (1)
+    {
+        addr_size = sizeof(clientAddress);
+        clientSocket = accept(serverSocket, (struct sockaddr *)&clientAddress, &addr_size);
+
+        // printf("Client Socket = %d\n", clientSocket);
+        syslog(LOG_INFO, "[+] Client Socket = %d\n", clientSocket);
+
+        totalBuffer = BUFFER_STD_SIZE;
+        currentSize = 0;
+
+        writebuffer = malloc(sizeof(char) * BUFFER_STD_SIZE);
+
+        // filePointer = fopen("dhiraj.txt", "w");
+
+        if (clientSocket < 0)
+        {
+            perror("[-] Accept Failed\n");
+            exit(-1);
+        }
+
+        // printf("Client Connected\n");
+        syslog(LOG_INFO, "Accepted connection from : %s\n", inet_ntoa(clientAddress.sin_addr));
+        // printf("Client Family : %d\n", clientAddress.sin_family);
+        // printf("Client Port : %d\n", clientAddress.sin_port);
+
+        bool exitLoop = true;
+
+        while (exitLoop)
+        {
+            recvClient = recv(clientSocket, receiveFromClient, BUFFER_STD_SIZE, 0);
+
+            if (recvClient == 0 || (strchr(receiveFromClient, '\n') != NULL))
+            {
+                syslog(LOG_INFO, "Packet Completed\n");
+                exitLoop = false;
+            }
+
+            if ((totalBuffer - currentSize) < recvClient)
+            {
+                totalBuffer += recvClient;
+                writebuffer = (char *)realloc(writebuffer, sizeof(char) * totalBuffer);
+            }
+
+            memcpy(writebuffer + currentSize, receiveFromClient, recvClient);
+            currentSize += recvClient;
+        }
+
+        bytesToBeWritten = write(fd, writebuffer, currentSize);
+        lseekValue = lseek(fd, 0, SEEK_SET);
+
+        // int fseekValue = fseek(filePointer, 0, SEEK_END);
+
+        // printf("lseekValue  = %d\n", lseekValue);
+
+        // printf("Bytes To be Written = %d\n", bytesToBeWritten);
+
+        readBufferSize += currentSize;
+
+        syslog(LOG_INFO, "Current Size = %d\n", currentSize);
+        syslog(LOG_INFO, "Read Bytes = %d\n", readBufferSize);
+
+        readFromFile = (char *)malloc(sizeof(char) * readBufferSize);
```

Comment:
@dhirajbennadi what assumption are you making here about the size of all packets received?